Automattic/woocommerce-payments
# Storefront integration for Multi-Currency

Fixes #2264 

#### Changes proposed in this Pull Request

**edited by @ismaeldcom**

- Create the new `StorefrontIntegration` class to implement the functionality of adding the switcher widget to Storefront theme through settings.
- Check if the option `Add a currency switcher to the Storefront theme` is enabled (on by default) and the used theme is Storefront or a child of it.
- Add the widget on the breadcrumb section of the store, as per final decision on the issue.

@jessepearson already did a good implementation, thanks! I simplified the code to make it more maintainable, removing the need for custom CSS code for each Storefront child theme. Using the available wrapper instead of adding a new one.
Removed unneeded code.

Updated the option description to mention that the widget will be added to breadcrumb section. **Hotel** theme for example have breadcrumb removed, so will not show the widget.

As per @LevinMedia comment below. I modified the settings to only show the switcher checkbox when a compatible theme is active and the message `A currency switcher is available in your widgets. Configure now` as fallback. You can see both of them in the screenshots below.

#### Screenshots
Here are some pics to see how it looks on different Storefront based themes.

|Storefront|
|:--:|
|<img width="772" alt="storefront" src="https://user-images.githubusercontent.com/7670276/126311569-7b3430ac-b436-49cf-8031-e3a1002978d7.png">|
|Arcade|
|<img width="774" alt="arcade" src="https://user-images.githubusercontent.com/7670276/126311570-7c23feb7-3680-4d4f-80f3-d3ccdb207c4b.png">|
|Galleria|
|<img width="775" alt="galleria" src="https://user-images.githubusercontent.com/7670276/126311574-a1486526-1a51-42d5-aae8-a0600e47ec8d.png">|
|Boutique|
|<img width="729" alt="boutique" src="https://user-images.githubusercontent.com/7670276/126311575-0ee17f10-111f-41f8-bb63-28e092bc3263.png">|

|Store settings with compatible theme|
|:--:|
|<img width="775" alt="Screenshot 2021-07-21 at 08 55 43" src="https://user-images.githubusercontent.com/7670276/126451656-168bcd9e-8727-436d-b92d-69b32d433fc3.png">|
|Store settings with incompatible theme|
|<img width="782" alt="Screenshot 2021-07-21 at 08 56 17" src="https://user-images.githubusercontent.com/7670276/126451747-d9a9dbaf-8387-4406-a554-f37d5d2612d0.png">|


#### Testing instructions
- Navigate to **WooCommerce > Settings > Multi-currency**, toggle `Add a currency switcher to the Storefront theme on breadcrumb section`. Check that the widget got included (or not) on the breadcrumb section of the store.
- Check that it works with different Storefront themes.
- It shouldn't be included on themes not based on Storefront or those without the breadcrumb section.
- The store settings should be like the attached screenshots when switching themes.

-------------------
